src/main/java/org/radarcns/stream/phone/PlayStoreLookup.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2017 King's College London and The Hyve
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.radarcns.stream.phone;
+
+import java.io.IOException;
+import org.apache.kafka.common.cache.LRUCache;
+import org.jsoup.HttpStatusException;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A Google Play Store lookup backed by a cache.
+ */
+public final class PlayStoreLookup {
+    private static final Logger log = LoggerFactory.getLogger(PlayStoreLookup.class);
+    private static final String URL_PLAY_STORE_APP_DETAILS = "https://play.google.com/store/apps/details?id=";
+    private static final String CATEGORY_ANCHOR_SELECTOR = "a.document-subtitle.category";
+
+    private final long cacheTimeoutSeconds;
+    // Do not cache more than 1 million elements, for memory consumption reasons
+    private final LRUCache<String, AppCategory> categoryCache;
```

Comment:
We considered persisting it, but it would introduce another layer of complexity (file locking, when to store, how to determine item age after persisting). This way it stays stateless, which will not have a big detriment as long as the service stays online for a reasonable amount of time.